### PR TITLE
qemu_guest_agent: Add new case for installing qga via installer

### PR DESCRIPTION
**Pull request by 6-dehan**
check qemu-ga.service function after installing via
virtio-win-guest-installer.

ID: 2101738
Signed-off-by: demeng <demeng@redhat.com>

### CODE REVIEW

**Comment by 6-dehan** [comment]
@vivianQizhu could you please help to review? 
(1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_driver_update_via_installer_tool.q35: STARTED
 (1/1) Host_RHEL.m9.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_driver_update_via_installer_tool.q35: PASS (306.02 s)

**Comment by 6-dehan** [comment]
@leidwang Could you please help to review this patch as well? appreciate it.

**Comment by vivianQizhu** [comment]
@6-dehan Hi, Please move from 'Draft' to 'Ready for review' if it is ready. And I actually found several different scripts has exactly same implementation on the first few steps. We have discussed with @leidwang and @menli820 that we would like to have a unique module for adding extra function test to those existed cases, sorry that not let you know earlier, I assumed that qga might be different but seems I was wrong about that, checking the existed ones obviously qga also share the same logic. Let's wait for @menli820's PR or we can discuss if you would like to take it. Thanks.

File: qemu/tests/qemu_guest_agent.py [hunk 1]
```diff
@@ -4274,6 +4277,40 @@ def gagent_check_run_qga_as_program(self, test, params, env):
...
+
+        vm = env.get_vm(params["main_vm"])
+        session = vm.wait_for_login()
+
+        uninstall_gagent(session, test, gagent_uninstall_cmd)
+        session = vm.reboot(session)
+        install_test_with_screen_on_desktop(vm, session, test, run_install_cmd,
+                                            installer_pkg_check_cmd,
+                                            copy_files_params=params)
+        win_installer_test(session, test, params)
```
Comment (vivianQizhu) [inline qemu/tests/qemu_guest_agent.py:4307]: Hi @6-dehan We will finally remove gagent related steps on those installer cases, so here we can have the test like your current implementation, but do we need this `win_installer_test` in gagent test?

**Comment by vivianQizhu** [comment]
@6-dehan Is this ready for review?

File: qemu/tests/qemu_guest_agent.py (hunk 1 above)
Comment (6-dehan) [reply qemu/tests/qemu_guest_agent.py:4307]: oh, after checking on that function further, I think it's better to remove it, it's not related to gagent.

Comment (6-dehan) [reply qemu/tests/qemu_guest_agent.py:4307]: yeah, I think we need  

> @6-dehan Is this ready for review?

nope, I don't add the content that we talked on our last meeting. I'll work on it asap.

**Comment by 6-dehan** [comment]
> @6-dehan Is this ready for review?

nope, I don't add the content that we talked on our last meeting. I'll work on it asap.

**Comment by 6-dehan** [comment]
Hi @vivianQizhu, could you please continue to help to review this? 
(1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_driver_update_via_installer_tool.q35: PASS (226.28 s)

File: qemu/tests/qemu_guest_agent.py [hunk 2]
```diff
@@ -4274,6 +4276,47 @@ def gagent_check_run_qga_as_program(self, test, params, env):
...
+    @error_context.context_aware
+    def gagent_check_driver_update_via_installer_tool(self, test, params, env):
+        """
+        Acceptance installer test:
+
+        1) Check the qga version that installed from iso.
+        2) Remove existed qga.
+        3) Install driver via virtio-win-guest-tools.exe.
+        4) Check the qga version after installing from installer.exe.
+        5) Verify the qemu-ga function works well.
```
[the remark this answers is not in the document]
Comment (6-dehan) [reply qemu/tests/qemu_guest_agent.py:4288]: qemu-ga function testing just can be verified by sending qga command and get the corresponding results. as the following code, line4308~4313:

**Review by vivianQizhu — APPROVED** [review]

**Pull request merged**